api/storage_service.cc

```diff
@@ -1359,16 +1370,30 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
         return make_ready_future<json::json_return_type>(0);
     });
 
-    ss::get_ownership.set(r, [&ss] (std::unique_ptr<http::request> req) {
+    ss::get_ownership.set(r, [&ctx, &ss] (std::unique_ptr<http::request> req) {
+        if (any_of_keyspaces_use_tablets(ctx)) {
+            throw httpd::bad_param_exception("storage_service/ownership cannot be used when a keyspace uses tablets");
+        }
+
         return ss.local().get_ownership().then([] (auto&& ownership) {
             std::vector<storage_service_json::mapper> res;
             return make_ready_future<json::json_return_type>(map_to_key_value(ownership, res));
         });
     });
 
     ss::get_effective_ownership.set(r, [&ctx, &ss] (std::unique_ptr<http::request> req) {
-        auto keyspace_name = req->param["keyspace"] == "null" ? "" : validate_keyspace(ctx, req->param);
-        return ss.local().effective_ownership(keyspace_name).then([] (auto&& ownership) {
+        auto keyspace = req->param["keyspace"] == "null" ? "" : validate_keyspace(ctx, req->param);
```

Comment:
Why is `keyspace_name` changed to `keyspace`?
Better reduce the churn if this is unnecessary.

Reply:
Fixed :+1: